scripts/cli/handler/parse.py

```diff
@@ -0,0 +1,377 @@
+    print 'with crashes :\'('
+  elif proc.returncode == 0:
+    print 'successfully! :-D'
+  else:
+    print 'with error! :-('
+
+  return proc.returncode
+
+def __collect_sql(projectdir):
+  '''This function finds all .sql files under the CODECOMPASS_HOME and creates
+  two other .sql files to the project directory. One is the parse.sql which is
+  the concatenation of all .sql files except the comments, index creations and
+  alter commands. The other one is after.sql which is the concatenation of all
+  .sql files except the comments, table creations and drop commands.'''
+
+  def remove_by_regex(s, begin, end):
+    return re.sub(begin + '([\s\S]*?)' + end + '\n?', '', s)
+
+  def sql_remove_drops(s):
+    return remove_by_regex(s, 'DROP', ';')
+
+  def sql_remove_create_tables(s):
+    return remove_by_regex(s, 'CREATE TABLE', ';')
+
+  def sql_remove_comments(s):
+    return remove_by_regex(s, '\/\*', '\*\/')
+
+  def sql_remove_constraints(s):
+    return remove_by_regex(s, 'ALTER TABLE', ';')
+
+  def sql_remove_indexes(s):
+    return remove_by_regex(s, 'CREATE INDEX', ';')
+
+  sql = ''
+  for curr in os.walk(join(os.environ['CODECOMPASS_HOME'], 'share', 'codecompass', 'sql')):
+    for filename in curr[2]:
+      if filename.endswith('.sql') and 'postgresql' not in curr[0]:
+        with open(join(curr[0], filename), 'r') as f:
+          sql += f.read()
+
+  before = sql_remove_constraints(sql_remove_indexes(sql_remove_comments(sql)))
+  after  = sql_remove_drops(sql_remove_create_tables(sql_remove_comments(sql)))
+
+  with open(join(projectdir, 'parse.sql'), 'w') as f:
+    f.write(before)
+
+  with open(join(projectdir, 'after.sql'), 'w') as f:
+    f.write(after)
+
+def __create_database_and_clean(args):
+  #--- Helper variables ---#
+
+  projectdir = join(args.workdir, args.name)
+  dblogfile = join(projectdir, 'dblog')
+
+  #--- Database operations ---#
+
+  print __current_time(), 'Creating new database...'
+
+  __collect_sql(projectdir)
+
+  util.pgsql_cmd(projectdir, args, 'dropdb', args.name)
+
+  res = util.pgsql_cmd(
+    projectdir, args,
+    'createdb', '-E', 'SQL_ASCII', '-T', 'template0', '-l', 'C', args.name)
+
+  if res != 0:
+    sys.stderr.write(
+      'createdb returned with ' + str(res) + '!\nSee ' + dblogfile +
+      ' for details!\n')
+    sys.exit(1)
+
+  res = util.pgsql_cmd(
+    projectdir, args, 'psql', '-f', 'parse.sql', '-d', args.name)
+
+  if res != 0:
+    sys.stderr.write(
+      'Failed to create the database schema (psql returned with ' + str(res) +
+      ')!\nSee ' + dblogfile + ' for details!\n')
+    sys.exit(1)
+
+  #--- Clean ---#
+
+  try:
+    shutil.rmtree(join(projectdir, 'data'))
+    os.remove(join(projectdir, 'codechecker_output'))
+    os.remove(join(projectdir, 'parselog'))
+    os.remove(join(projectdir, 'buildlog'))
+    for filename in os.listdir(projectdir):
+      if filename.startswith('core') or filename.startswith('crash_'):
+        os.remove(join(projectdir, filename))
```

Comment:
Without iteration how we can remove the files?
```py
  for filename in os.listdir(projectdir):
      if filename not in ['build.json', 'db.log', 'after.sql']:
        os.remove(join(projectdir, filename))
```
Moreover we cannot remove the db.log and the after.sql script too. It will use it in later code.